### PR TITLE
[OSDOCS-7969]: Update OSD docs to include deployment using Red Hat Marketplace

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.14+
Issue:

https://issues.redhat.com/browse/OSDOCS-7969
Link to docs preview:

https://70006--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_install_access_delete_cluster/creating-a-gcp-cluster#osd-create-cluster-rhm-gcp-account_osd-creating-a-cluster-on-gcp
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by svmrh** [comment]
Since this is the least popular deployment option for OSD, can we have QE validate the steps? @xueli181114: Can you please assign someone from the QE team familiar with OSD deployment on the Red Hat Marketplace to try out the steps and review this PR? 

@mletalie: 
- There are some steps that don't apply to the Red Hat Marketplace. E.g. "4.a Click the Review Google Terms and Agreements link." was verify specific to OSD deployment with GCP Marketplace. 
- Since this is the least popular option, can we move this section as the last deployment option on GCP (i.e. after "Creating a cluster on GCP with a Red Hat cloud account")? 
- I haven't tried this deployment option myself, and therefore unclear about the steps. We should wait for QE to comment.

**Comment by mletalie** [comment]
> Since this is the least popular deployment option for OSD, can we have QE validate the steps? @xueli181114: Can you please assign someone from the QE team familiar with OSD deployment on the Red Hat Marketplace to try out the steps and review this PR?
> 
> @mletalie:
> 
> * There are some steps that don't apply to the Red Hat Marketplace. E.g. "4.a Click the Review Google Terms and Agreements link." was verify specific to OSD deployment with GCP Marketplace.
Yes, good point. I had not gone through the steps yet, and I need to do that. At this point just getting the template going, so to speak. 

> * Since this is the least popular option, can we move this section as the last deployment option on GCP (i.e. after "Creating a cluster on GCP with a Red Hat cloud account")?
Yes, I can do that. 
> * I haven't tried this deployment option myself, and therefore unclear about the steps. We should wait for QE to comment.
Understood. I will certainly try the full deployment and have QE run through it was well. 

In regard to the opening paragraph, is there anything you would suggest to add to it. Maybe something that differentiates from the other options in a positive light...ie, why it would be a good choice for some users, what its stand out features are, etc? Not sure if that makes sense but I am at a loss for additional info. Hope that question makes sense. Thanks! 

Thanks @svmrh, appreciate the info. Please see comments above.

**Comment by mletalie** [comment]
Reviewers: The majority of this topic was pulled from already existing topic https://docs.openshift.com/dedicated/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-cluster-gcp-account_osd-creating-a-cluster-on-gcp as many of the steps to create the cluster are similar. The differences are specific to the selected market place only (in this case, Red Hat Marketplace).

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+The VPC owner of the host project has 30 days to grant the listed permissions before the cluster creation fails.
+For information about Shared VPC permissions, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#migs-service-accounts[Provision Shared VPC].
+====
++
+. If you opted to install the cluster in an existing GCP VPC, provide your *Virtual Private Cloud (VPC) subnet settings* and select *Next*.
+You must have created the Cloud network address translation (NAT) and a Cloud router. See the additional resources for information about Cloud NATs and Google VPCs.
++
+[NOTE]
+====
+You must ensure that your VPC is configured with a public and a private subnet for each availability zone that you want the cluster installed into. If you opted to use PrivateLink, only private subnets are required.
```
Comment (jianli-wei) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:120]: Suggest to remove this statement, which is not true for installing cluster on GCP. 

FYI To install a cluster in an existing GCP VPC, the VPC is expected to have a subnet for control-plane machine and a subnet for compute machines.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+====
++
+[NOTE]
+====
+If you are installing a cluster into a Shared VPC, the VPC name and subnets are shared from the host project.
```
Comment (jianli-wei) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:125]: Suggest to replace "the VPC name and subnets are **shared** from the host project" with "the VPC name and subnets are from the host project".

**Review by jianli-wei — CHANGES_REQUESTED** [review]

**Comment by mletalie** [comment]
**Reviewers**: The majority of this topic was pulled from already existing topic https://docs.openshift.com/dedicated/osd_install_access_delete_cluster/creating-a-gcp-cluster.html#osd-create-cluster-gcp-account_osd-creating-a-cluster-on-gcp as many of the steps to create the cluster are similar. The main differences are specific to the selected market place only (in this case, Red Hat Marketplace).

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
++
+[IMPORTANT]
+====
+Once you complete the steps within the cluster configuration wizard and click *Create Cluster*, the cluster will go into the "Installation Waiting" state. At this point, you must contact the VPC owner of the host project, who must assign the dynamically-generated service account the following roles: *Computer Network Administrator*, *Compute Security Administrator*, and *DNS Administrator*.
+The VPC owner of the host project has 30 days to grant the listed permissions before the cluster creation fails.
+For information about Shared VPC permissions, see link:https://cloud.google.com/vpc/docs/provisioning-shared-vpc#migs-service-accounts[Provision Shared VPC].
+====
++
+. If you opted to install the cluster in an existing GCP VPC, provide your *Virtual Private Cloud (VPC) subnet settings* and select *Next*.
+You must have created the Cloud network address translation (NAT) and a Cloud router. See the additional resources for information about Cloud NATs and Google VPCs.
```
Comment (ShaunaDiaz) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:116]:
```suggestion
You must have created the Cloud network address translation (NAT) and a Cloud router. See the "Additional resources" section for information about Cloud NATs and Google VPCs.
```
Usually call that thusly.

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+====
+
+. On the *Cluster update strategy* page, configure your update preferences:
+.. Choose a cluster update method:
+** Select *Individual updates* if you want to schedule each update individually. This is the default option.
+** Select *Recurring updates* to update your cluster on your preferred day and start time, when updates are available.
++
+[NOTE]
+====
+You can review the end-of-life dates in the update lifecycle documentation for {product-title}. For more information, see _{product-title} update lifecycle_.
```
Comment (ShaunaDiaz) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:165]: Can a link be dropped in for the lifecycle?

File: modules/osd-create-cluster-rhm-gcp-account.adoc [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+====
++
+.. Provide administrator approval based on your cluster update method:
+** Individual updates: If you select an update version that requires approval, provide an administrator’s acknowledgment and click *Approve and continue*.
+** Recurring updates: If you selected recurring updates for your cluster, provide an administrator’s acknowledgment and click *Approve and continue*. {cluster-manager} does not start scheduled y-stream updates for minor versions without receiving an administrator’s acknowledgment.
++
+For information about administrator acknowledgment, see xref:./../upgrading/osd-upgrading-cluster-prepare.adoc#upgrade-49-acknowledgement_osd-updating-cluster-prepare[Administrator acknowledgment when upgrading to OpenShift 4.9].
```
Comment (ShaunaDiaz) [inline modules/osd-create-cluster-rhm-gcp-account.adoc:172]: Is there a supported version link? 4.11+?

**Review by ShaunaDiaz — COMMENTED** [review]
Looks great! Just a few questions.

**Pull request merged**